### PR TITLE
Update: File-upload package

### PR DESCRIPTION
**Pull request by hiteshchaudhari-everestek**
- Added ref integration in the file-upload package for both forwardRef and ref.
- Updated the key used to upload the file to the s3 bucket.
- Used 'bucketFolderPath' prop to for the key through which the file will be uploaded to a specific folder in s3 bucket.

### CODE REVIEW

File: packages/file-upload/index.js [hunk 1]
```diff
@@ -131,28 +139,38 @@ class FileUpload extends PureComponent {
...
+  }),
 };
 
 FileUpload.defaultProps = {
   bucket: null,
+  bucketFolderPath: null,
   customPrefix: { private: '', protected: '', public: '' },
   error: false,
   existingFiles: [],
+  forwardRef: null,
```
Comment (cadejscroggins) [inline packages/file-upload/index.js:168]: Why do we need forwardRef & ref? Why not just ref?

File: packages/file-upload/index.js [hunk 2]
```diff
@@ -131,28 +139,38 @@ class FileUpload extends PureComponent {
...
   onRemoveComplete: PropTypes.func,
   onUploadComplete: PropTypes.func,
+  ref: PropTypes.shape({
+    current: PropTypes.object,
+  }),
 };
 
 FileUpload.defaultProps = {
   bucket: null,
+  bucketFolderPath: null,
```
Comment (cadejscroggins) [inline packages/file-upload/index.js:164]: What's the difference between this and `customPrefix`?

File: packages/file-upload/index.js (hunk 1 above)
Comment (hiteshchaudhari-everestek) [reply packages/file-upload/index.js:168]: When we need to add a reference to the child component, we cannot directly pass it as ref. We need to pass as forwardRef. So, 'forwardRef' meaning we are forwarding a ref to the child component and then we are using that 'forwardRef' as a prop to assign it to the actual component we need to assign

File: packages/file-upload/index.js (hunk 2 above)
Comment (hiteshchaudhari-everestek) [reply packages/file-upload/index.js:164]: I guess here, we can pass bucket folder path in 'customPrefix' to upload a file to a certain folder in a bucket. I'll update this file to remove the 'bucketFolderPath' prop used as a key.

File: packages/file-upload/index.js (hunk 1 above)
Comment (cadejscroggins) [reply packages/file-upload/index.js:159]: Ah ok. Thanks.

**Pull request merged**